### ISSUE TITLE
Finding the right microprocessor

### ISSUE THREAD

**Issue by nihlaeth** [Humanity4all/AsetNiop4Teensy#17]
We were planning on moving from the teensy to the atmega328 microprocessor, mainly because of the cost. However, the ftdi serial2usb ic turns out to be incapable of presenting as a human input device to the os. Alternatives to this ftdi chip cost about the same as an arduino uno equivalent, at which point it'd probably be easier to just use that. Only... It needs a firmware upgrade to present as a human input device, and probably a non-standard library, which may not be very reliable. At the same time we ran out of pins (the LEDs take up a lot, and then there's 10 switches and the trackball and sd reader). Looking at the larger boards, prices increase rapidly. The Arduino Due (equivalent) is still cheaper than the teensy, but it has no eeprom, and an external eeprom is not standard equipment for arduino tinkerers. It's hard to find and it's doubtful there are arduino libraries for it. There's always multiplexers, to solve the pin shortage, only it costs time to poll them. They can only be used for pins that are to be read or written to infrequently (like the LEDs and chip select pins for the spi interface).

So, to recap:
- Teensy (3.0) - relatively expensive and we need a multiplexer, but good usb hid support
- Arduino Due - cheaper than teensy, enough pins and good usb hid support, but no eeprom
- Arduino Uno (r3) - cheap (as long as you buy an 'equivalent' and not the real deals), needs a multiplexer, usb hid support requires a bit of hacking
- Bare Atmega328 - extremely cheap, still needs a multiplexer, needs an ic to connect over usb (which makes it about the same price as an arduino uno).

I'm leaning towards the uno equivalent. Added benefit is that it's widely supported and available worldwide.
